.github/workflows/macos.yaml

```
@@ -151,8 +151,14 @@ jobs:
         S3_OMNIBUS_CACHE_BUCKET: "dd-ci-datadog-agent-omnibus-cache-build-stable"
         S3_OMNIBUS_CACHE_ANONYMOUS_ACCESS: "true"
         SIGN: ${{ secrets.ENABLE_SIGN }}
+        USE_CACHING_PROXY_RUBY: "true"
```

I think this should be:
```suggestion
        USE_CACHING_PROXY_RUBY: ${{ secrets.ENABLE_CACHING_PROXY }}
```
(with the `ENABLE_CACHING_PROXY` secret set to `true`) for the same reason we have

```
  SIGN: ${{ secrets.ENABLE_SIGN }}
```

above: if an Action runs without permissions to access secrets (eg. dependabot PRs), the Action should still succeed.